scrapy_poet/utils.py

```diff
@@ -121,27 +84,31 @@ def is_response_going_to_be_used(request, spider):
     if is_callback_using_response(callback):
         return True
 
-    plan, _ = build_plan(callback, None)
-    for provider in get_providers(plan):
+    for provider in get_providers(callback):
         if is_provider_using_response(provider):
             return True
 
     return False
 
 
-def get_providers(plan: andi.Plan):
-    for obj, _ in plan:
-        provider = providers.get(obj)  # type: ignore
-        if not provider:
-            continue
+def get_providers(callback):
+    for argument, possible_types in andi.inspect(callback).items():
+        for cls in possible_types:
+            if is_injectable(cls):
+                yield from get_providers(cls)
+                continue
 
-        yield provider
+            provider = providers.get(cls)
+            if not provider:
+                continue
+
+            yield provider
 
 
-def build_plan(callback, response
+def build_plan(callback, instances
```

Comment:
:lipstick:  This is only styling preference, but better a single long line than a break..

Reply:
Yeah, this is an issue that's more likely to happen with the use of type annotations.

Also prefer long lines in cases like this (function definitions). Let me try to make it work and see if our pipeline is complaining.

Reply:
Done.